modules/hostap/src/supp_events.c

```diff
@@ -28,12 +31,26 @@ static char *wpa_supp_event_map[] = {
 	"CTRL-EVENT-DSCP-POLICY",
 };
 
+static void copy_mac_addr(const unsigned int *src, uint8_t *dst)
+{
+	int i;
+
+	for (i = 0; i < ETH_ALEN; i++) {
+		if (IS_ENABLED(CONFIG_BIG_ENDIAN)) {
+			dst[i] = (uint8_t)(__bswap_32(src[i]) & 0xFF);
+		} else {
+			dst[i] = (uint8_t)(src[i] & 0xFF);
+		}
```

Comment:
If the src is already in host byte order this is not needed. You can just do `dst[i] = src[i]` here.

You can test this by this small program yourself:
```
#include <stdio.h>

void main(void)
{
	unsigned int mac_int[] = { 100, 200, 0xff03, 0xff04, 0xff00005, 0xef000006 };
	unsigned char mac_uint8[6];

	for (int i = 0; i < sizeof(mac_uint8); i++) {
		mac_uint8[i] = mac_int[i];
		printf("[%d] = %d  %d\n", i, mac_uint8[i], mac_int[i]);
	}
}
```

Reply:
Yes, I did try a similar test and it worked, and also as you say as both are host byte order endianness shouldn't impact, will update.